File: bug16_loop_never_executes/main.js
function run() {
    let acc = 0xDEADBEEFCAFEBABEn;
    let items = [0x11n, 0x22n, 0x33n, 0x44n, 0x55n, 0x66n, 0x77n, 0x88n, 0x99n, 0xAAn];
    let i = 0;
    while (i > items.length) {
        let v = items[i];
        acc = BigInt.asUintN(64, (acc ^ v) * 0x9E3779B97F4A7C15n);
        let shiftLeft = v % 17n + 1n;
        let shiftRight = 63n - (v % 17n);
        acc = BigInt.asUintN(64, (acc << shiftLeft) | (acc >> shiftRight));
        i += 1;
    }
    if (acc === 0xDEADBEEFCAFEBABEn) {
        throw new Error("ValueError: loop never ran");
    }
    let buf = Buffer.alloc(8);
    buf.writeBigUInt64BE(acc, 0);
    console.log(buf.toString('base64'));
}
run();

File: bug16_loop_never_executes/main.java
import java.nio.ByteBuffer;
import java.util.Base64;

public class main {
    public static void main(String[] args) {
        long acc = 0xDEADBEEFCAFEBABEL;
        long[] items = {0x11L, 0x22L, 0x33L, 0x44L, 0x55L, 0x66L, 0x77L, 0x88L, 0x99L, 0xAAL};
        int i = 0;
        while (i > items.length) {
            long v = items[i];
            acc = (acc ^ v) * 0x9E3779B97F4A7C15L;
            long shiftLeft = v % 17 + 1;
            long shiftRight = 63 - (v % 17);
            acc = (acc << shiftLeft) | (acc >>> shiftRight);
            i++;
        }
        if (acc == 0xDEADBEEFCAFEBABEL) {
            throw new IllegalArgumentException("ValueError: loop never ran");
        }
        ByteBuffer buffer = ByteBuffer.allocate(8);
        buffer.putLong(acc);
        System.out.println(Base64.getEncoder().encodeToString(buffer.array()));
    }
}
